File: src/domain/TripState.ts
// src/domain/TripState.ts
export type TripState = "READY" | "ASSIGNED" | "EN_ROUTE" | "COMPLETED" | "CANCELED";

export const TripState = {
  READY: "READY" as TripState,
  ASSIGNED: "ASSIGNED" as TripState,
  EN_ROUTE: "EN_ROUTE" as TripState,
  COMPLETED: "COMPLETED" as TripState,
  CANCELED: "CANCELED" as TripState,
};

File: src-tauri/src/domain/trip_state.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
pub enum TripState {
    Draft,
    Ready,
    EnRoute,
    Completed,
    Cancelled,
}

impl ToString for TripState {
    fn to_string(&self) -> String {
        match self {
            TripState::Draft => "DRAFT",
            TripState::Ready => "READY",
            TripState::EnRoute => "EN_ROUTE",
            TripState::Completed => "COMPLETED",
            TripState::Cancelled => "CANCELLED",
        }
        .into()
    }
}
